### PR TITLE
Add tests for functions 6-10

### PR DESCRIPTION
**Pull request by dmitrymarokhonov**

### CODE REVIEW

File: src/add.spec.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+describe('Test suite for "add"-function', () => {
+    it('Should return 5 on input 3 and 2', () => {
+        expect(add(3, 2)).toBe(5)
+    })
+    it('Should return 5 on input 3 and 2', () => {
+        expect(add(3, '2')).not.toBe(5)
+    })
+    it('Should return -37 on input 3 and -40', () => {
+        expect(add(3, -40)).toBe(-37)
```
Comment (juliusstoerrle) [inline src/add.spec.js]: Do you think a test with two negative values is sensible?

File: src/add.spec.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+// The add function should be tested to ensure that the addition of the provided integers & 
+// floating-point numbers provides the mathematically correct result. 
+
+describe('Test suite for "add"-function', () => {
+    it('Should return 5 on input 3 and 2', () => {
+        expect(add(3, 2)).toBe(5)
+    })
+    it('Should return 5 on input 3 and 2', () => {
+        expect(add(3, '2')).not.toBe(5)
```
Comment (juliusstoerrle) [inline src/add.spec.js:11]: I like the test in general, but note that the jsdoc comment on the add function asks for the number type not string. So if it fails this could be considered still correct - even through I would expect an error to be thrown. For me we can keep it with the argumentation that a user will likely pass it string or a user input as string without being aware of the type expectation.

File: src/capitalize.spec.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        it('Should leave "-hello" as without changing"', () => {
+            expect(capitalize('-hello')).toBe('-hello')
+        })
+        it('Should leave "-_?" as it is', () => {
+            expect(capitalize('-_?')).toBe('-_?')
+        })
+        it('Should leave "/\fs" as it is', () => {
+            expect(capitalize('/\fs')).toBe('/\fs')
+        })
+
```
Comment (juliusstoerrle) [inline src/capitalize.spec.js:24]: How does capitalize deal with special letters like ö, ä, à, â etc.? This could be another testcase I think,

File: src/chunk.spec.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+// The chunk function should be tested to ensure all elements 
+// of the original array are returned in correctly sized chunks. 
+
+describe('Test suite for "chunk"-function', () => {
+    describe('Test simple array', () => {
+        // const arr = ['a', 'b']
+        it("Should return [['a', 'b'], ['c', 'd']], on input (['a','b', 'c','d'], 2)", () => {
+            console.log(chunk(['a', 'b', 'c', 'd'], 2));
+            // expect(chunk(['a', 'b', 'c', 'd'], 2)).toBe([['a', 'b'], ['c', 'd']])
```
Comment (juliusstoerrle) [inline src/chunk.spec.js]: I would leave the failing test in, for me its ok if the testsuite fails in the end. Since that actually shows the true state of the libary. We could also fix the actual functions, but I don't think that is neccesary at this stage of the process.

Did you find a input with which the function deals correctly?

File: src/get.spec.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        })
+        it('Should return \'undefined\' on non existing element', () => {
+            expect(get(arrStr, '[3]')).toBe(undefined)
+        })
+        it('Should return \'null\' on non existing element with default value specified as \'null\'', () => {
+            expect(get(arrStr, '[3]', null)).toBe(null)
+        })
+    })
+
+    describe('Complex object testing with inner array and object', () => {
```
Comment (juliusstoerrle) [inline src/get.spec.js]: The below testcases look good to me, however the seccond parameter could accept an array too according to the jsdoc. Maybe you can add similar cases with array instead of string? `'nestedObj1.a' -> ['nestedObj1','a']`

File: src/isEmpty.spec.js [hunk 6]
```diff
@@ -1,5 +1,81 @@
...
+      sampleMap.clear()
+      expect(isEmpty(sampleMap)).toBe(true)
+    })
+
+  })
+  describe('Prototype tests', () => {
+    function Foo() { }
+    function Bar() { }
+    Bar.prototype = Object.create(Foo.prototype)
+    it('Should return on prototype', () => {
```
Comment (juliusstoerrle) [inline src/isEmpty.spec.js]:
```suggestion
    it('should return true when the object is empty', () => {
```

File: src/isEmpty.spec.js [hunk 7]
```diff
@@ -1,5 +1,81 @@
...
+      expect(isEmpty(new Bar())).toBe(true)
```
Comment (juliusstoerrle) [inline src/isEmpty.spec.js:81]: I think we should have a case for a non empty object too.

File: src/isEmpty.spec.js [hunk 8]
```diff
@@ -1,5 +1,81 @@
...
+    sampleMap.set('a', 'a1')
+    sampleMap.set('b', 'b1')
+
+    sampleSet.add(1)
+    sampleSet.add('a')
+    it('Should return false on non-empty Set', () => {
+      expect(isEmpty(sampleSet)).toBe(false)
+    })
+    it('Should return true on empty Set', () => {
+      sampleSet.clear()
```
Comment (juliusstoerrle) [inline src/isEmpty.spec.js]: Jest could have a mode where they run tests in parallel or something. Therefore, I think the test result should not depend on the test order. I suggest to clone sampleSet and sampleMap before clearing them.

```suggestion
      emptySet = new Set(sampleSet).clear();
```

File: src/chunk.spec.js (hunk 4 above)
Comment (juliusstoerrle) [reply src/chunk.spec.js]: Also I would remove the console.log statement once it is not needed since it disturbs the test runner output.

File: src/capitalize.spec.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import capitalize from "./capitalize"
+
+
+describe('Test suite for "capitalize" funtion', () => {
+    it('Should convert "hello" to "Hello"', () => { expect(capitalize('hello')).toBe('Hello') })
+    it('Should apply only to first word if passed several words', () => {
+        expect(capitalize('several words written')).toBe('Several words written')
+    })
+    describe('Testing string with non-letter characters', () => {
```
Comment (juliusstoerrle) [inline src/capitalize.spec.js]: It was not obvious why you are testing this. I guess you want to ensure that special character at the start of a string stay unchanged. I would rephrase the name or add a comment I think.
```suggestion
    describe('with strings starting with special characters', () => {
```

File: src/capitalize.spec.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import capitalize from "./capitalize"
+
+
+describe('Test suite for "capitalize" funtion', () => {
```
Comment (juliusstoerrle) [inline src/capitalize.spec.js]:
```suggestion
describe('capitalize', () => {
```

File: src/add.spec.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
+import add from "./add"
+
+// The add function should be tested to ensure that the addition of the provided integers & 
+// floating-point numbers provides the mathematically correct result. 
+
+describe('Test suite for "add"-function', () => {
```
Comment (juliusstoerrle) [inline src/add.spec.js]: I would usually just write "add". To create a sentence with the it() after. e.g. "add should return 4..."
```suggestion
describe('add', () => {
```

File: src/chunk.spec.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import chunk from './chunk'
+
+// The chunk function should be tested to ensure all elements 
+// of the original array are returned in correctly sized chunks. 
+
+describe('Test suite for "chunk"-function', () => {
```
Comment (juliusstoerrle) [inline src/chunk.spec.js]:
```suggestion
describe('chunk', () => {
```

File: src/get.spec.js [hunk 13]
```diff
@@ -0,0 +1,66 @@
+import get from './get'
+// The get function shall be tested to ensure different levels (1,2,3) and 
+// kind (array, object) property paths return the designated property. 
+
+describe('Testing suite for "get"-function', () => {
```
Comment (juliusstoerrle) [inline src/get.spec.js]:
```suggestion
describe('get', () => {
```

File: src/isEmpty.spec.js [hunk 14]
```diff
@@ -1,5 +1,81 @@
 import isEmpty from './isEmpty';
+// The tests for the isEmpty function must include checks for null, 
+// objects, collection, map, or set each of which shall return true 
+// if size or length respectively return 0 or contain no properties.  
 
-test('identifies null as empty', () => {
-  expect(isEmpty(null)).toBe(true);
-});
\ No newline at end of file
+describe('Test suite for "isEmpty" function', () => {
```
Comment (juliusstoerrle) [inline src/isEmpty.spec.js]:
```suggestion
describe('isEmpty', () => {
```

File: src/isEmpty.spec.js [hunk 15]
```diff
@@ -1,5 +1,81 @@
...
+    it('Should return false for non-empty Map', () => {
+      expect(isEmpty(sampleMap)).toBe(false)
+    })
+    it('Should return true for empty Map', () => {
+      sampleMap.clear()
+      expect(isEmpty(sampleMap)).toBe(true)
+    })
+
+  })
+  describe('Prototype tests', () => {
```
Comment (juliusstoerrle) [inline src/isEmpty.spec.js]:
```suggestion
  describe('on object/prototype values', () => {
```

**Review by juliusstoerrle — COMMENTED** [review]
@dmitrymarokhonov Your test cases look pretty good to me. I added some comments inline, only like 5 or 6 regarding the tests itself the rest more on naming matters. Overall I think its pretty good.

File: src/isEmpty.spec.js (hunk 7 above)
Comment (dmitrymarokhonov) [reply src/isEmpty.spec.js:81]: done

File: src/add.spec.js (hunk 1 above)
Comment (dmitrymarokhonov) [reply src/add.spec.js]: test case removed

File: src/add.spec.js (hunk 2 above)
Comment (dmitrymarokhonov) [reply src/add.spec.js:11]: description updated

File: src/capitalize.spec.js (hunk 3 above)
Comment (dmitrymarokhonov) [reply src/capitalize.spec.js:24]: added test cases with Ö & Ä ('scandinavian') letters
